### ISSUE TITLE
Need `scheme:default` to change link color

### ISSUE THREAD

**Issue by leonardehrenfried** [squidfunk/mkdocs-material#4594]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

In order to override the primary color with
```
theme:
    palette:
        primary: blue
```

you also need the following:
```
        scheme: default
```
even though it should already be set by default.

### Expected behaviour

I expected the link colour to be set to blue.

### Actual behaviour

The link colour remains at indigo.

### Steps to reproduce

1. Set primary colour
2. Don't set a theme
3. See that link colour doesn't match primary colour.

### Package versions

- Python: `3.10.7`
- MkDocs: `mkdocs 1.4.2`
- Material: `8.5.8`

### Configuration

```yaml
theme:
    name: material
    features:
        - navigation.tabs
        - toc.integrate
    palette:
        primary: blue
        accent: blue
```

### System information

- Operating system: Fedora Linux
- Browser: Firefox

**Comment by squidfunk** [comment]
Fixed in 074a0c86cd6276133ff04f54e4f6cad26b6ad8b3.

**Comment by squidfunk** [comment]
Released as part of 8.5.9.
